mobile_verifier/src/heartbeats/mod.rs

```diff
@@ -741,6 +780,116 @@ pub async fn clear_heartbeats(
     Ok(())
 }
 
+/// A cache for previous valid (or invalid) WiFi heartbeat locations
+#[derive(Clone)]
+pub struct LocationCache {
+    pool: PgPool,
+    locations: Arc<Cache<PublicKeyBinary, Option<LastLocation>>>,
+}
+
+impl LocationCache {
+    pub fn new(pool: &PgPool) -> Self {
+        let locations = Arc::new(Cache::new());
+        let locations_clone = locations.clone();
+        tokio::spawn(async move {
+            locations_clone
+                .monitor(4, 0.25, std::time::Duration::from_secs(60 * 60 * 24))
+                .await
+        });
+        Self {
+            pool: pool.clone(),
+            locations,
+        }
+    }
+
+    async fn fetch_from_db_and_set(
+        &self,
+        hotspot: &PublicKeyBinary,
+    ) -> anyhow::Result<Option<LastLocation>> {
+        let last_location: Option<LastLocation> = sqlx::query_as(
+            r#"
+            SELECT location_validation_timestamp, lat, lon
+            FROM wifi_heartbeats
+            WHERE location_validation_timestamp IS NOT NULL
+                AND location_validation_timestamp >= $1
+            ORDER BY location_validation_timestamp DESC
+            LIMIT 1
+            "#,
+        )
+        .bind(Utc::now() - Duration::hours(12))
```

Comment:
so the intent here is if the radio has not have a valid location_validation_timestamp within last 12 hours then default it.  But why 12 hours and not 24 or other?

Reply:
I don't know, that was the requirement given to me